burn-import/src/burn/node/constant.rs

```diff
@@ -1,72 +1,153 @@
+            ConstantValue::Int64(_) => quote! { i64 },
+            ConstantValue::Tensor(tensor_type, _) => {
+                let ty = tensor_type.ty();
+                quote! { #ty}
+            }
         }
     }
     pub fn val_tokens(&self) -> TokenStream {
         match self {
-            ConstantValue::Int(val) => quote! { #val },
-            ConstantValue::Float(val) => quote! { #val },
-            ConstantValue::Bool(val) => quote! { #val },
+            ConstantValue::Float32(val) => quote! { #val },
+            ConstantValue::Int64(val) => quote! { #val },
+            ConstantValue::Tensor(_, _) => {
+                panic!("Tensor constant is not assignable.")
+            }
         }
     }
 }
 
-impl ConstantNode {
-    pub fn new(name: String, value: ConstantValue) -> Self {
-        let output_ty = OtherType::new(name.clone(), value.ty_tokens());
-
+impl<PS: PrecisionSettings> ConstantNode<PS> {
+    pub fn new(name: String, value: ConstantValue<PS>, output: Type) -> Self {
         Self {
             name,
             value,
-            output_ty,
+            output,
+        }
+    }
+    pub fn constant_value_into_type(&self) -> Type {
+        let name = Ident::new(self.name.as_str(), Span::call_site());
+        match &self.value {
+            ConstantValue::Float32(_) => Type::Scalar(ScalarType {
+                name,
+                kind: ScalarKind::Float32,
+            }),
+            ConstantValue::Int64(_) => Type::Scalar(ScalarType {
+                name,
+                kind: ScalarKind::Int64,
+            }),
+            ConstantValue::Tensor(tensor_type, _) => Type::Tensor(tensor_type.clone()),
         }
     }
 }
 
-impl<PS: PrecisionSettings> NodeCodegen<PS> for ConstantNode {
+impl<PS: PrecisionSettings> NodeCodegen<PS> for ConstantNode<PS> {
     fn output_types(&self) -> Vec<Type> {
-        vec![Type::Other(&self.output_ty)]
+        vec![self.output.clone()]
     }
 
     fn input_types(&self) -> Vec<Type> {
         vec![]
     }
 
+    fn field_type(&self) -> Option<Type> {
+        match &self.value {
+            ConstantValue::Tensor(tensor_type, _) => Some(Type::Tensor(tensor_type.clone())),
+            _ => None,
+        }
+    }
+
+    fn field_init(&self, with_record: bool) -> Option<TokenStream> {
+        match &self.value {
+            ConstantValue::Tensor(tensor_type, _) => {
+                let ty = tensor_type.ty();
+                let name = Ident::new(self.name.as_ref(), Span::call_site());
+                let shape = tensor_type.clone().shape.unwrap().to_tokens();
+                let dim = tensor_type.clone().dim.to_tokens();
+
+                if with_record {
+                    Some(quote! {
+                        let #name = record.#name;
+                    })
+                } else {
+                    Some(quote! {
+                        let #name: #ty = Tensor::<B, #dim>::zeros(#shape);
```

Comment:
I think tensor fields should be `Param<Tensor<B, D>>`  even if constant. When constant, we should just call `set_require_grad(false)` during the init. This would avoid having to change the record type for `Tensor<B, D>`.

Reply:
I think this is appropriate when we need the tensor to be included in the final model record but behaves as a constant (Constant model parameter VS just a constant tensor).

Reply:
I think we can make it work without changing Tensor record. I'll try it.

Reply:
With your help, we removed and now using Param with grad disabled in place of tensor module field. This allows to save/load weights after a finetuning if needed.